### ISSUE TITLE
Implement `BoundedStorable` for more data types

### ISSUE THREAD

**Issue by 0kenx** [dfinity/stable-structures#92]
```
 help: the following other types implement trait `BoundedStorable`:
              ()
              (A, B)
```
Would be nice to implement this for (A, B, C...) in accordance with candid.

**Comment by 0kenx** [comment]
Also for Vec<T> where T is BoundedStorable

**Comment by ielashi** [comment]
> Would be nice to implement this for (A, B, C...) in accordance with candid.

I agree. We don't rely on candid for the (A, B) encoding though and neither should we for (A, B, C). If you have the cycles maybe this is something you'd like to contribute? :)

> Also for Vec<T> and [T; N], where T is BoundedStorable and N: usize

`Vec<T>` is not bounded, so we cannot implement `BoundedStorable` for it. We already have a `BoundedStorable` implementation for `[T; N]` (see [here](https://github.com/dfinity/stable-structures/blob/29fba6bb3ec15cdb93b82c1b5f912b59ab746394/src/storable.rs#L295))
